File: 1695-maximum-erasure-value/1695-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int n=nums.size();
        unordered_map<int,int>mp;
        int sum=0;
        int maxsum=0;
        int j=0;
        for(int i=0;i<n;i++){
            while(mp.find(nums[i])!=mp.end()){
                mp[nums[j]]--;
                sum=sum-nums[j];
                if(mp[nums[j]]==0){
                    mp.erase(nums[j]);
                }
                j++;
            }
            sum+=nums[i];
            maxsum=max(sum,maxsum);
            mp[nums[i]]++;
        }
        return maxsum;
    }
};

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        int n=nums.length;
        int sum=0;
        int j=0;
        int maxscore=Integer.MIN_VALUE;
        HashMap<Integer,Integer>mp=new HashMap<>();
        for(int i=0;i<n;i++){
            while(mp.containsKey(nums[i])){
                sum-=nums[j];
                mp.remove(nums[j]);
                j++;
            }
            mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
            sum+=nums[i];
            maxscore=Math.max(maxscore,sum);
        }

        return maxscore;
    }
}